File: fish-feed-backend/app/models/pond_dto.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import datetime


class PondBase(BaseModel):
    name: str
    location: Optional[str] = None
    size_sq_m: Optional[float] = None  # Pond size in square meters
    depth_m: Optional[float] = None  # Pond depth in meters
    stocking_date: Optional[datetime] = None  # Date when stocked
    stocking_density: Optional[int] = (
        None  # Number of fish/shrimp per square meter
    )
    water_type: Optional[str] = None  # e.g., freshwater, brackish, marine
    status: Optional[str] = "active"  # Pond status like active, inactive, drained
    is_active: Optional[bool] = True


class PondCreate(PondBase):
    organization_id: Optional[UUID] = None


class PondUpdate(BaseModel):
    name: str
    location: Optional[str] = None
    size_sq_m: Optional[float] = None  # Pond size in square meters
    depth_m: Optional[float] = None  # Pond depth in meters
    stocking_date: Optional[datetime] = None  # Date when stocked
    stocking_density: Optional[int] = (
        None  # Number of fish/shrimp per square meter
    )
    water_type: Optional[str] = None  # e.g., freshwater, brackish, marine
    status: Optional[str] = "active"  # Pond status like active, inactive, drained
    is_active: Optional[bool] = True


class PondDto(PondBase):
    id: UUID
    organization_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: fish-feed-frontend/src/dto/pond.dto.ts
import type { Species } from "./species.dto";

export interface Pond {
  id?: string;
  organization_id: string;
  name: string;
  location?: string;
  size_sq_m: number;
  depth_m: number;
  species?: Species[]; // assuming you have a Species type
  stocking_date?: string; // ISO string format
  stocking_density?: number;
  water_type?: string;
  status?: "active" | "maintenance" | "harvested" | string;
  is_active?: boolean;
  additional_attributes?: Record<string, any>;
  created_at?: string;
  updated_at?: string;
}
